src/mbgl/map/map_context.cpp

```diff
@@ -113,6 +113,8 @@ void MapContext::setStyleURL(const std::string& url) {
 
         if (res.status == Response::Successful) {
             loadStyleJSON(res.data, base);
+        } else if (res.status == Response::NotFound && (styleURL.find("mapbox://") != std::string::npos)) {
```

Comment:
`std::string::npos` is `-1` and matches against the entire string, yes. Check out b385ef269808a2be132de5a9ad41b5c8aa34d099, which only proceeds on matches that start at position `0` — i.e., the beginning of the string.